lib/rspec/matchers/built_in/all.rb

```diff
@@ -50,7 +50,13 @@ def match(_expected, _actual)
         def index_failed_objects
           actual.each_with_index do |actual_item, index|
             cloned_matcher = matcher.clone
-            matches = cloned_matcher.matches?(actual_item)
+
+            begin
+              matches = cloned_matcher.matches?(actual_item)
+            rescue StandardError
+              matches = nil
```

Comment:
It's non obvious why this is here, and generally it's not something we want to do, so reasoning is required before sneaking this under the radar